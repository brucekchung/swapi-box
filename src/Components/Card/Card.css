.Card {
  color: white;
  height: 300px;
  width: 200px;
  border: 1px solid white;
  margin: 20px;
  text-align: center;
}

.favorite {
  color: grey;
}