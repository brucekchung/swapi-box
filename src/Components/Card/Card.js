import React from 'react'
import Entry from '../Entry/Entry'
import './Card.css'

const Card = ({ data, type, favorite }) => {
  const entries = Object.entries(data)
  const name = entries.shift()
  const last = entries.slice(entries.length - 1)
  const isFavoriteCard = () => {
    if(last[0][1] === true) {
      return 'favorite'
    } else return ''
  }
  
  if (last[0][0] === 'favorite') entries.pop()

  const info = entries.map((entry, index) => 
    ( <Entry key={type + index} entry={entry} /> ))

  return (
    <div className={`Card ${isFavoriteCard()}`} id={name[1]}>
      <button className="favorite-btn" onClick={favorite} />
      <h2>{name[1]}</h2>
      { info }
    </div>
  )
}

export default Card