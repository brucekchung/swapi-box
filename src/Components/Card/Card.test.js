import React from 'react'
import { shallow } from 'enzyme'
import Card from './Card'

describe('Card', () => {
  const props = {
    key: 14,
    data: {
      "model": "Digger Crawler", 
      "name": "Sand Crawler",
    },
    type: 'testCard',
    favorite: () => {},
  }

  it('should match the snapshot', () => {
    const wrapper = shallow(<Card {...props}/>)
    
    expect(wrapper).toMatchSnapshot()
  })
})