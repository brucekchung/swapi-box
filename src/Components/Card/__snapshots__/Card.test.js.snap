// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={
        Object {
            "model": "Digger Crawler",
            "name": "Sand Crawler",
          }
    }
    favorite={[Function]}
    type="testCard"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="favorite-btn"
          onClick={[Function]}
/>,
        <h2>
          Digger Crawler
</h2>,
        Array [
          <Entry
            entry={
                        Array [
                                    "name",
                                    "Sand Crawler",
                                  ]
            }
/>,
        ],
      ],
      "className": "Card ",
      "id": "Digger Crawler",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "favorite-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Digger Crawler",
        },
        "ref": null,
        "rendered": "Digger Crawler",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": "testCard0",
        "nodeType": "function",
        "props": Object {
          "entry": Array [
            "name",
            "Sand Crawler",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="favorite-btn"
            onClick={[Function]}
/>,
          <h2>
            Digger Crawler
</h2>,
          Array [
            <Entry
              entry={
                            Array [
                                          "name",
                                          "Sand Crawler",
                                        ]
              }
/>,
          ],
        ],
        "className": "Card ",
        "id": "Digger Crawler",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "favorite-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Digger Crawler",
          },
          "ref": null,
          "rendered": "Digger Crawler",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": "testCard0",
          "nodeType": "function",
          "props": Object {
            "entry": Array [
              "name",
              "Sand Crawler",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
